casper-v2/src/main/scala/coop/rchain/casper/v2/core/DependencyGraph.scala

```diff
@@ -0,0 +1,57 @@
+package coop.rchain.casper.v2.core
+import cats.effect.Sync
+import cats.syntax.all._
+
+trait DependencyGraph[F[_], M] {
+
+  /** Each DAG message is justified by set of other messages. */
+  def justifications(message: M): F[List[M]]
+
+  /**
+    * Parent is a justification which cannot be viewed through other justifications.
```

Comment:
I'm not sure If we should call this parent - its first observation of the message in the whole view, parent might be quite misleading imo.